### PR TITLE
TG PORT: Folder TGUI

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request

ports https://github.com/tgstation/tgstation/pull/57320. Credits to him for being awsome

As a note TG uses `update_apperance`, I changed all instande of `update_apperance ` to `update_icon` and hope a maintainer knows what its supposed to be

ALso on L 105 missing TRAIT_HANDS_BLOCKED since its TG exclusive. If anyone knows the bee version let me know

## Why It's Good For The Game

Folders are now based and red pilled instead of dumb and cringe. 

## Changelog
:cl:
tweak: Fancy new UI for folders
/:cl:

### CODE REVIEW

**Comment by ItsTisme** [comment]
Will do the clipboard PR aswel once he actuely finished it

**Comment by ItsTisme** [comment]
Smh ofcourse I run into a error because upstream linter check had a bug that ours doesnt.

**Comment by Archanial** [comment]
`update_appearance` is a wrapper proc that contains `update_name` `update_desc` and `update_icon`.
It's not exactly only `update_icon`.
See here for the PR that changed it: https://github.com/tgstation/tgstation/pull/55468
We should really port this but 558 files changed :pensive:

File: code/modules/paperwork/folders.dm [hunk 1]
```diff
@@ -6,119 +6,114 @@
...
 
-/obj/item/folder/white
-	desc = "A white folder."
-	icon_state = "folder_white"
+/obj/item/folder/proc/rename(mob/user)
+	if(!user.is_literate())
+		to_chat(user, "<span class='notice'>You scribble illegibly on the cover of [src]!</span>")
+		return
 
+	var/inputvalue = stripped_input(user, "What would you like to label the folder?", "Folder Labelling", "", MAX_NAME_LEN)
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
	var/inputvalue = stripped_input(user, "What would you like to label the folder?", "Folder Labelling", null, MAX_NAME_LEN)
```

File: code/modules/paperwork/folders.dm [hunk 2]
```diff
@@ -6,119 +6,114 @@
...
+		return
+
+	if(user.canUseTopic(src, BE_CLOSE))
+		name = "folder[(inputvalue ? " - '[inputvalue]'" : null)]"
+
+/obj/item/folder/proc/remove_item(obj/item/Item, mob/user)
+	if(istype(Item))
+		Item.forceMove(user.loc)
+		user.put_in_hands(Item)
+		to_chat(user, "<span class='notice'>You remove [Item] from [src].</span>")
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
/obj/item/folder/proc/remove_item(obj/item/I, mob/user)
	if(istype(I))
		I.forceMove(user.loc)
		user.put_in_hands(I)
		to_chat(user, "<span class='notice'>You remove [I] from [src].</span>")
```

File: code/modules/paperwork/folders.dm [hunk 3]
```diff
@@ -6,119 +6,114 @@
...
+		update_icon()
+
+/obj/item/folder/AltClick(mob/user)
+	..()
+	if(contents)
+		remove_item(contents[1], user)
 
+/obj/item/folder/update_overlays()
+	. = ..()
+	if(contents.len)
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
	if(LAZYLEN(contents))
```

File: code/modules/paperwork/folders.dm [hunk 4]
```diff
@@ -6,119 +6,114 @@
...
+	if(istype(src, /obj/item/folder/syndicate))
+		data["theme"] = "syndicate"
+	data["bg_color"] = "[bg_color]"
+	data["folder_name"] = "[name]"
+
+	data["contents"] = list()
+	data["contents_ref"] = list()
+	for(var/Content in src)
+		data["contents"] += "[Content]"
+		data["contents_ref"] += "[REF(Content)]"
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
	for(var/content in src)
		data["contents"] += "[content]"
		data["contents_ref"] += "[REF(content)]"
```

File: code/modules/paperwork/folders.dm [hunk 5]
```diff
@@ -6,119 +6,114 @@
...
-/obj/item/folder/syndicate/blue
-	icon_state = "folder_sblue"
+	if(.)
+		return
 
-/obj/item/folder/syndicate/blue/Initialize()
-	. = ..()
-	new /obj/item/documents/syndicate/blue(src)
-	update_icon()
+	if(usr.stat != CONSCIOUS)
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
	if(usr.incapacitated())
```
That covers more stuff than just conscious

File: code/modules/paperwork/folders.dm [hunk 6]
```diff
@@ -6,119 +6,114 @@
...
 
-/obj/item/folder/syndicate/mining/Initialize()
-	. = ..()
-	new /obj/item/documents/syndicate/mining(src)
-	update_icon()
+	switch(action)
+		// Take item out
+		if("remove")
+			var/obj/item/Item = locate(params["ref"]) in src
+			remove_item(Item, usr)
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
			var/obj/item/I= locate(params["ref"]) in src
			remove_item(I, usr)
```

File: code/modules/paperwork/folders.dm [hunk 7]
```diff
@@ -6,119 +6,114 @@
...
+			. = TRUE
+		// Inspect the item
+		if("examine")
+			var/obj/item/Item = locate(params["ref"]) in src
+			if(istype(Item))
+				usr.examinate(Item)
```
Comment (Archanial) [inline code/modules/paperwork/folders.dm]:
```suggestion
			var/obj/item/I = locate(params["ref"]) in src
			if(istype(I))
				usr.examinate(I)
```

File: code/modules/paperwork/folders.dm (hunk 1 above)
Comment (ItsTisme) [reply code/modules/paperwork/folders.dm]: I assume behind the scenes they both are null, but I cant help but get feeling something wont render properly

like the case wether its empty string or actuely nothing(null)

**Comment by ItsTisme** [comment]
> `update_appearance` is a wrapper proc that contains `update_name` `update_desc` and `update_icon`.
> It's not exactly only `update_icon`.
> See here for the PR that changed it: [tgstation/tgstation#55468](https://github.com/tgstation/tgstation/pull/55468)
> We should really port this but 558 files changed 😔

So I gueuse its "fine for now" sorta deal? mainly icon matters

**Comment by ItsTisme** [comment]
I did the fix finaly please merge I dont wana do it again

**Comment by ItsTisme** [comment]
why must you let this happen again git

**Review by francinum — APPROVED** [review]

**Pull request merged**